### PR TITLE
propagate_positions: Bugfix + Better documentation

### PR DESCRIPTION
**Pull request by erezsh**
Addresses issue #1293

### CODE REVIEW

File: lark/lark.py [hunk 1]
```diff
@@ -87,7 +87,9 @@ class LarkOptions(Serialize):
     transformer
             Applies the transformer to every parse tree (equivalent to applying it after the parse, but faster)
     propagate_positions
-            Propagates (line, column, end_line, end_column) attributes into all tree branches.
+            Propagates positional attributes into the 'meta' attribute of all tree branches.
+            Sets attributes: (line, column, end_line, end_column, start_pos, end_pos,
+                              container_line, container_column, container_end_line, container_end_column)
```
Comment (MegaIng) [inline lark/lark.py:92]: Is it documented anywhere what the `container_*` versions mean?

Comment (erezsh) [reply lark/lark.py:92]: Hmm it's not.. Just this comment in the code -

```python
            # Calculate positions while the tree is streaming, according to the rule:
            # - nodes start at the start of their first child's container,
            #   and end at the end of their last child's container.
            # Containers are nodes that take up space in text, but have been inlined in the tree.
```

File: lark/tree.py [hunk 2]
```diff
@@ -47,7 +47,9 @@ class Tree(Generic[_Leaf_T]):
         data: The name of the rule or alias
         children: List of matched sub-rules and terminals
         meta: Line & Column numbers (if ``propagate_positions`` is enabled).
-            meta attributes: line, column, start_pos, end_line, end_column, end_pos
+            meta attributes: (line, column, end_line, end_column, start_pos, end_pos,
+                              container_line, container_column, container_end_line, container_end_column)
+            container_* attributes consider all symbols, including those that have been inlined in the tree.
```
Comment (MegaIng) [inline lark/tree.py:52]: I think an example of the difference might be nice. I have a hard time understanding where this makes a difference.

**Review by MegaIng — APPROVED** [review]

File: lark/tree.py (hunk 2 above)
Comment (erezsh) [reply lark/tree.py:52]: Added.

**Pull request merged**